parser-typechecker/src/Unison/TermPrinter.hs

```diff
@@ -1097,65 +1106,104 @@ calcImports im tm = (im', render $ getUses result)
--- syntax that get a call to `calcImports` in `pretty0`.  AST nodes that do a calcImports in
--- pretty0, in order to try and emit a `use` statement, need to be emitted also by this
--- function, otherwise the `use` statement may come out at an enclosing scope instead.
-immediateChildBlockTerms :: (Var vt, Var v) => Term2 vt at ap v a -> [Term2 vt at ap v a]
+allInSubBlock ::
+  (Var v, Ord v) =>
+  Term3 v PrintAnnotation ->
+  Prefix ->
+  Suffix ->
+  Int ->
+  Bool
+allInSubBlock tm p s i =
+  let found = concat $ ABT.find finder tm
+      result = any (/= tm) found
+      tr =
+        const id $ trace
+          ( "\nallInSubBlock("
+              ++ show p
+              ++ ", "
+              ++ show s
+              ++ ", "
+              ++ show i
+              ++ "): returns "
+              ++ show result
+              ++ "\nInput:\n"
+              ++ show tm
+              ++ "\nFound: \n"
+              ++ show found
+              ++ "\n\n"
+          )
+   in tr result
+  where
+    getUsages t =
+      annotation t
+        |> usages
+        |> Map.lookup s
+        |> fmap (Map.lookup p)
+        |> join
+        |> fromMaybe 0
+    finder t =
+      let result =
+            let i' = getUsages t
+             in if i' < i
+                  then ABT.Prune
+                  else
+                    let found = filter hit $ immediateChildBlockTerms t
+                     in if (i' == i) && (not $ null found)
+                          then ABT.Found found
+                          else ABT.Continue
+          children =
+            concat
+              ( map (\t -> "child: " ++ show t ++ "\n") $
+                  immediateChildBlockTerms t
+              )
+          tr =
+            const id $ trace
+              ( "\nfinder: returns "
+                  ++ show result
+                  ++ "\n  children:"
+                  ++ children
+                  ++ "\n  input: \n"
+                  ++ show t
+                  ++ "\n\n"
+              )
+       in tr $ result
+    hit t = (getUsages t) == i
+
+-- Return any blockterms at or immediately under this term. Has to match the
+-- places in the syntax that get a call to `calcImports` in `pretty0`.
+-- AST nodes that do a calcImports in pretty0, in order to try and emit a `use`
+-- statement, need to be emitted also by this function, otherwise the `use`
+-- statement may come out at an enclosing scope instead.
+immediateChildBlockTerms ::
+  (Var vt, Var v) => Term2 vt at ap v a -> [Term2 vt at ap v a]
 immediateChildBlockTerms = \case
-    Handle' handler body -> [handler, body]
-    If' _ t f -> [t, f]
-    LetBlock bs _ -> concat $ map doLet bs
-    Match' scrute branches ->
-      if isDestructuringBind scrute branches then [scrute]
-      else concat $ map doCase branches
-    _ -> []
+  Handle' handler body -> [handler, body]
+  If' _ t f -> [t, f]
+  LetBlock bs e -> concatMap doLet bs ++ handleDelay e
```

Comment:
Added the case where `e` (the expression at the end of the `let` block) is a `'let` block.